File: C++/Access.cpp
#include<iostream>
using namespace std;

///////////////////////////////////////////////////////////////
//
// Class Name : Demo
// Members :    int i (public), int j (private), int k (protected)
// Description: Demonstrates access specifiers in C++.
// Author:      Samruddh Shivkumar Birajdar
//
///////////////////////////////////////////////////////////////

class Demo {
    public:
        int i;          // Public member
    private:
        int j;          // Private member
    protected:
        int k;          // Protected member

    public:
        // Constructor
        Demo() {
            i = 0;
            j = 0;
            k = 0;
        }

        // Display method to show values of all members
        void display() {
            cout << "Value of i: " << i << endl;
            cout << "Value of j: " << j << endl;
            cout << "Value of k: " << k << endl;
        }
};

///////////////////////////////////////////////////////////////
//
// Application demonstrating access specifiers
//
///////////////////////////////////////////////////////////////

int main() {
    Demo dobj;

    // Input section
    cout << "Enter value for i: ";
    cin >> dobj.i;

    // Output section
    cout << "Displaying values through class method:" << endl;
    dobj.display(); // Allowed: method has access to all members

    cout << "Accessing public member directly: " << dobj.i << endl;

    // cout << dobj.j; // ❌ Not Allowed (private)
    // cout << dobj.k; // ❌ Not Allowed (protected)

    return 0;
}

File: Java/Access.java
///////////////////////////////////////////////////////////////
//
// Class Name : Demo
// Members :    public int i, private int j, protected int k
// Methods :    Constructor, Display()
// Description: Explains access specifier ranges in Java.
// Author:      Samruddh Shivkumar Birajdar
//
///////////////////////////////////////////////////////////////

class Demo {
    public int i;
    private int j;
    protected int k;

    public Demo() {
        i = 0;
        j = 0;
        k = 0;
    }

    public void Display() {
        System.out.println("Value of i: " + i); // Allowed
        System.out.println("Value of j: " + j); // Allowed
        System.out.println("Value of k: " + k); // Allowed
    }
}

///////////////////////////////////////////////////////////////
//
// Application : Shows which members are accessible in main and inheritance.
//
///////////////////////////////////////////////////////////////

public class Access {
    public static void main(String A[]) {
        Demo dobj = new Demo();
        dobj.Display();

        System.out.println("Value of i: " + dobj.i); // Allowed
        // System.out.println("Value of j: " + dobj.j); // Not Allowed (private)
        // System.out.println("Value of k: " + dobj.k); // Allowed only in inheritance, not here
    }
}
